pkg/prometheus/promcfg.go

```diff
@@ -770,6 +770,11 @@ func (cg *ConfigGenerator) GenerateServerConfiguration(
 		cfg = append(cfg, cg.generateRemoteReadConfig(remoteRead, store))
 	}
 
+	// OTLP config
+	if cpf.OTLP != nil && cg.version.GTE(semver.MustParse("2.54.0")) {
```

Comment:
Should we do `cg.WithMinimumVersion(...)`?
Or we could also reject the object at `resource_selector.go`. 
WDYT? 
CC: @simonpasquier

Reply:
In this case, I'd be in favor of failing the reconciliation indeed if the Prometheus version doesn't match the requirement. One idea would be to validate the requirements in NewConfigGenerator and return an error if necessary.

Reply:
Thanks @mviswanathsai @simonpasquier .  If the Prometheus version doesn't match the requirement, return error.